restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/RestcommCallsTool.java

```diff
@@ -367,4 +376,42 @@ public JsonArray getCallRecordings(String deploymentUrl, String username, String
 
         return jsonArray;
     }
+    
+    public String setGateWay(String deploymentUrl, String username, String authToken, String friend, String uName, String password, String proxy,
+        boolean register, String ttl) {
+
+        Client jerseyClient = Client.create();
+        jerseyClient.addFilter(new HTTPBasicAuthFilter(username, authToken));
+
+        String url = getGateWayUrl(deploymentUrl, username);
+
+        WebResource webResource = jerseyClient.resource(url);
+
+        MultivaluedMap<String, String> params = new MultivaluedMapImpl();
+        params.add("Register", String.valueOf(register));
+        if (friend != null) {
+            params.add("FriendlyName", friend);
+        }
+        if (uName != null) {
+            params.add("UserName", uName);
+        }
+        if (password != null) {
+            params.add("Password", password);
+        }
+        if (proxy != null) {
+            params.add("Proxy", proxy);
+        }
+        if (ttl != null) {
+            params.add("TTL", ttl);
+        }
+
+        String response = null;
+        try {
+            response = webResource.accept(MediaType.APPLICATION_XML).post(String.class, params);
```

Comment:
why are we using XML here, most of the testsuite is using JSON as response format to ease parsing later...

Reply:
updated follow comment.